src/ifp/src/InitFloorplan.cc

```diff
@@ -480,11 +480,21 @@ void InitFloorplan::makeTracks()
   for (auto layer : block_->getDataBase()->getTech()->getLayers()) {
     if (layer->getType() == dbTechLayerType::ROUTING
         && layer->getRoutingLevel() != 0) {
-      makeTracks(layer,
-                 layer->getOffsetX(),
-                 layer->getPitchX(),
-                 layer->getOffsetY(),
-                 layer->getPitchY());
+      if (layer->getTechLayerPitchRules().empty()) {
+        makeTracks(layer,
+                   layer->getOffsetX(),
+                   layer->getPitchX(),
+                   layer->getOffsetY(),
+                   layer->getPitchY());
+      } else {
+        auto pitch_rule = layer->getTechLayerPitchRules().begin();
+        makeTracksNonUniform(layer,
+                             layer->getOffsetX(),
+                             pitch_rule->getPitchX(),
+                             layer->getOffsetY(),
+                             pitch_rule->getPitchY(),
+                             pitch_rule->getFirstLastPitch());
```

Comment:
If more than 1 can be defined, why only generate one?

Reply:
According to the definition Matt sent me, this rule is supposed to be maximum 1 per layer. It allows to generate more than 1 because I used the codeGenerator, but never more than 1 is supposed to be created .

Reply:
I'm also unclear if only one can be defined the layer->getTechLayerPitchRules() should return a single instance.

Reply:
All the methods regarding the class dbTechLayerPitchRule were generated using the code generator, and the layer->getTechLayerPitchRules() made by the code generator returns a table. I didn't tried to change anything done by the code generator. @osamahammad21 is there a way to make it return just the class instance?

Reply:
Unfortunately 1 to 1 relations are not supported yet in the odb code generator.

Reply:
It seems like maybe this would be better implemented outside of the code generator then? Otherwise we will have to go back and change this data type later which might not be so easy

Reply:
Looking at this I think it would be simpler to add a _first_last_pitch field to dbTechLayer (like pitch_x/y).  It would have a no-value (eg -1).  Simple hasFirstLastPitch() and getFirstLastPitch() methods on dbTechLayer would suffice.  It doesn't have to be a separate object.